src/openrct2/core/String.hpp

```diff
@@ -222,6 +222,10 @@ class CodepointView
                 GetNextCodepoint(&_str[_index], &nextch);
                 _index = nextch - _str.data();
             }
+            else
+            {
+                _index = _str.size();
+            }
```

Comment:
This is still problematic, the _index is out of bounds for one iteration before it calls the operator again and only then it would equal the end iterator. Use std::min on line 223 to ensure _index is always clamped.